### PR TITLE
 added support for populating platform resource map 

### PR DESCRIPTION
**Pull request by cdietrich**

### CODE REVIEW

**Comment by cdietrich** [comment]
new try with rebased base cc @kthoms @szarnekow @janhicken

**Comment by cdietrich** [comment]
[maven-xcore-sample.zip](https://github.com/eclipse/xtext-maven/files/2069497/maven-xcore-sample.zip) sample project for easier experiments

**Comment by ArneDeutsch** [comment]
Tested this on windows. Works well when "autoFillPlatformResourceMap" is enabled. Fails as expected when disabled. Works again with manual project mappings.

**Comment by cdietrich** [comment]
ping @szarnekow

File: org.eclipse.xtext.maven.plugin/src/main/java/org/eclipse/xtext/maven/XtextGenerator.java [hunk 1]
```diff
@@ -227,4 +268,50 @@ private void configureDefaults() {
...
+
+	/**
+	 * Adds the given file to EcorePlugin's platform resource map.
+	 * The file's name will be used as the map entry's key.
+	 *
+	 * @param file a file to register
+	 * @return the registered URI pointing to the given file
+	 * @see EcorePlugin#getPlatformResourceMap()
+	 */
+	private static URI addToPlatformResourceMap(final File file, final Log log) {
```
Comment (szarnekow) [inline org.eclipse.xtext.maven.plugin/src/main/java/org/eclipse/xtext/maven/XtextGenerator.java]: Why do we pass the log here but not into manuallyAddToPlatformResourceMap?

Comment (cdietrich) [reply org.eclipse.xtext.maven.plugin/src/main/java/org/eclipse/xtext/maven/XtextGenerator.java]: @szarnekow fixed. is caused by different authors

**Comment by szarnekow** [comment]
Nice!

**Comment by cdietrich** [comment]
thx @janhicken

**Pull request merged**